server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -393,28 +395,30 @@ public void requireBuffer(
       RequireBufferRequest request, StreamObserver<RequireBufferResponse> responseObserver) {
     String appId = request.getAppId();
     long requireBufferId;
-    if (StringUtils.isEmpty(appId)) {
-      // To be compatible with older client version
-      requireBufferId =
-          shuffleServer.getShuffleTaskManager().requireBuffer(request.getRequireSize());
-    } else {
-      requireBufferId =
-          shuffleServer
-              .getShuffleTaskManager()
-              .requireBuffer(
-                  appId,
-                  request.getShuffleId(),
-                  request.getPartitionIdsList(),
-                  request.getRequireSize());
-    }
-
     StatusCode status = StatusCode.SUCCESS;
-    if (requireBufferId == RequireBufferStatusCode.NO_BUFFER.statusCode()) {
-      status = StatusCode.NO_BUFFER;
-      ShuffleServerMetrics.counterTotalRequireBufferFailed.inc();
-    } else if (requireBufferId == RequireBufferStatusCode.NO_REGISTER.statusCode()) {
+    try {
+      if (StringUtils.isEmpty(appId)) {
+        // To be compatible with older client version
+        requireBufferId =
+            shuffleServer.getShuffleTaskManager().requireBuffer(request.getRequireSize());
+      } else {
+        requireBufferId =
+            shuffleServer
+                .getShuffleTaskManager()
+                .requireBuffer(
+                    appId,
+                    request.getShuffleId(),
+                    request.getPartitionIdsList(),
+                    request.getRequireSize());
+      }
+    } catch (NoRegisterException e) {
+      requireBufferId = -1;
       status = StatusCode.NO_REGISTER;
       ShuffleServerMetrics.counterTotalRequireBufferFailed.inc();
+    } catch (NoBufferException | NoBufferForHugePartitionException e) {
```

Comment:
This looks weird, why not introducing new stateCode of `NoBufferForHugePartitionException`